src/service/pokemonService.js

```diff
@@ -0,0 +1,83 @@
+import { getPokemonById, getTypeById } from "../api/pokemon"
+
+
+// example result of calling getPokemonById(1):
+// { id: 1, name: 'Bulbasaur', photoUrl: 'someurl', types: [ { id: 3, name: 'grass' }, { id: 6, name: 'poison' } ] }
+// example result of calling getTypeById(3):
+// { id: 3, name: 'grass' }
+
+
+// example of getNextQuestion('guessName')
+// { 
+//     question: { pokemonName: 'Bulbasaur', pokemonUrl: 'someUrl', }, 
+//     answers: [ 'answer1', 'answer2', 'answer3', 'answer4', 'answer5-forTypes', 'answer6-forTypes', ], 
+//     correctAnswer: 'Bulbasaur' 
+// }
+
+class Question {
+
+    constructor(mode) {
+        this.mode = mode;
+    }
+
+    //Random nr 1-151
+    randomPokemonId() {
+        const firstGenPokemonLimit = 152
+        let randomId = Math.floor(Math.random() * firstGenPokemonLimit)
+        return randomId;
+    }
+
+    answers(numOfAnswers = 4) {
+        let answersArr = [];
+        for (let i = 1; i <= numOfAnswers; i++) {
+            answersArr.push(this.randomPokemonId());
+        }
+        return answersArr
+    }
+
+    correctAnswer(numOfAnswers = 4) {
+        return Math.floor(Math.random() * numOfAnswers) + 1;
+    }
+
+    getNextQuestion() {
+        const answersList = this.answers();
+        const correctAnswerId = () => answersList[this.correctAnswer];
```

Comment:
Hmm, it looks like a bug - why it is needed to make `correctAswerId` a function? Talking about ` () => ` part.

Reply:
Yes, it was bug. I provided fix -> const correctAnswerId = answersList[this.correctAnswer];